Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -111,6 +114,53 @@ void TRDGlobalTracking::updateTimeDependentParams()
   mTracker->SetTPCVdrift(mTPCVdrift);
 }
 
+void TRDGlobalTracking::fillMCTruthInfo(const TrackTRD& trk, const o2::MCCompLabel& lblSeed, std::vector<o2::MCCompLabel>& lblContainer, const o2::dataformats::MCTruthContainer<o2::MCCompLabel>* trkltLabels) const
+{
+  // Check MC labels of the TRD tracklets attached to the track seed.
+  // Set TRD track label to the most frequent tracklet label.
+  // Fake flag is set if either not all TRD tracklets have most frequent label
+  // or if the seeding label is different from the most frequent TRD label.
+  // In case multiple tracklet labels occur most often we choose the one which matches the label of the seed, or,
+  // if that is not the case one of the most frequent labels is chosen arbitrarily
+  LOG(DEBUG) << "Checking seed with label: " << lblSeed;
+  std::unordered_map<o2::MCCompLabel, unsigned int> labelCounter;
+  int nTracklets = 0;
+  unsigned int maxOccurences = 0;
+  for (int iLy = 0; iLy < constants::NLAYER; ++iLy) {
+    auto trkltIndex = trk.getTrackletIndex(iLy);
+    if (trkltIndex == -1) {
+      // no tracklet in this layer
+      continue;
+    }
+    const auto& lblsTrklt = trkltLabels->getLabels(trkltIndex);
+    for (const auto lblTrklt : lblsTrklt) {
+      ++labelCounter[lblTrklt];
+      if (labelCounter[lblTrklt] > maxOccurences) {
+        maxOccurences = labelCounter[lblTrklt];
+      }
+    }
+  }
+  o2::MCCompLabel mostFrequentLabel;
+  for (const auto& [lbl, count] : labelCounter) {
+    LOG(DEBUG) << "Label " << lbl << " occured " << count << " times.";
+    if (count == maxOccurences) {
+      if (lblSeed.compare(lbl) >= 0) {
```

Comment:
`MCCompLabel::compare` checks also for the isFake flag of each label, which you don't need, it is faster to do 
````
if (lblSeed == lbl) { 
````

Reply:
Done